payments-core/src/main/java/com/stripe/android/googlepaylauncher/GooglePayPaymentMethodLauncherActivity.kt

```diff
@@ -87,70 +101,63 @@ internal class GooglePayPaymentMethodLauncherActivity : AppCompatActivity() {
         setFadeAnimations()
     }
 
-    private fun launchGooglePay(task: Task<PaymentData>) {
-        AutoResolveHelper.resolveTask(
-            task,
-            this,
-            LOAD_PAYMENT_DATA_REQUEST_CODE
-        )
-    }
-
-    @Deprecated("Deprecated in Java")
-    public override fun onActivityResult(
-        requestCode: Int,
-        resultCode: Int,
-        data: Intent?
-    ) {
-        super.onActivityResult(requestCode, resultCode, data)
-        if (requestCode == LOAD_PAYMENT_DATA_REQUEST_CODE) {
-            when (resultCode) {
-                RESULT_OK -> {
-                    onGooglePayResult(data)
-                }
-                RESULT_CANCELED -> {
-                    updateResult(
-                        GooglePayPaymentMethodLauncher.Result.Canceled
-                    )
-                }
-                AutoResolveHelper.RESULT_ERROR -> {
-                    val status = AutoResolveHelper.getStatusFromIntent(data)
-                    val statusMessage = status?.statusMessage.orEmpty()
+    private fun onGooglePayResult(taskResult: ApiTaskResult<PaymentData>) {
+        when (taskResult.status.statusCode) {
+            CommonStatusCodes.SUCCESS -> {
+                val result = taskResult.result
+                if (result != null) {
```

Comment:
ah it actually looks like this activity handles the taskResult.result being null _and_ it reports errors, both of which are not present in GooglePayLauncherActivity. If that's not intentional, should we try to share more code between these two classes?